protocols/encoding/decoder.go

```diff
@@ -0,0 +1,30 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package encoding
+
+// MetricsDecoder decodes bytes into data model.
+type MetricsDecoder interface {
```

Comment:
Should we instead of "decode" use "deserialize" or "unmarshal" instead?

Reply:
Will go with Serialize/Deserialize based on Bogdan's comment to name package `serializer`.